src/index.ts

```diff
@@ -82,10 +83,19 @@ export async function startNotarize(opts: NotarizeStartOptions): Promise<Notariz
       notarizeArgs.push('-itc_provider', opts.ascProvider);
     }
 
-    const result = await spawn('xcrun', notarizeArgs);
-    if (result.code !== 0) {
-      throw new Error(`Failed to upload app to Apple's notarization servers\n\n${result.output}`);
-    }
+    const result = await pRetry(
+      async () => {
+        const result = await spawn('xcrun', notarizeArgs);
+        if (result.code !== 0) {
```

Comment:
From https://www.npmjs.com/package/p-retry#pretryinput-options

> Returns a Promise that is fulfilled when calling input returns a fulfilled promise. If calling input returns a rejected promise, input is called again until the maximum number of retries is reached. It then rejects with the last rejection reason.

So as I understand it, it will reject with the error and abort, like before

Reply:
I don't see when the spawn will actually throw an error with the current implementation. With the current PR, I don't think the process will ever retry with the example in #33's issue summary.

Reply:
Spawn doesn't throw an error but this one will, right?
```javascript
        if (result.code !== 0) {
          throw new pRetry.AbortError(
            `Failed to upload app to Apple's notarization servers\n\n${result.output}`,
          );
        }
```

Reply:
`pRetry.AbortError` will not retry. From the docs:

> `pRetry.AbortError(error)`
>
> Abort retrying and reject the promise.

Reply:
Gotcha. I misunderstood. So then I think my implementation will do nothing useful compared to the original code. It will only retry other errors if any, but if code is not 0 it will just abort without retrying like befre